doc: describe the helm chart resource (#287)
open-component-model/ocm-controller: docs/helm_chart_resource.md

# Helm Chart Resources

This document describes the Helm Chart resource type and it's usage.

## From OCM to controller resource

Support for Helm chart based resources is made possible through OCM. Describing a Helm Chart based resource through OCM
component version might look something like this:

```yaml
components:
- name: github.com/open-component-model/helm-test
  version: "v1.0.0"
  provider:
    name: ocm.software
  resources:
  - name: charts
    type: helmChart
    version: 6.3.5
    input:
      type: helm
      version: 6.3.5
      path: charts/podinfo-6.3.5.tgz
```

> This resource then can be defined for the controller like this:

Reviewer: ```suggestion
This resource then can be accessed by the ocm-controller like this:
```
